cmake/modules/extensions.cmake

```diff
@@ -1559,6 +1559,71 @@ function(zephyr_syscall_header_ifdef feature_toggle)
   endif()
 endfunction()
 
+# Verify blobs fetched using west. If the sha256 checksum isn't valid, a warning/
+# fatal error message is printed (depends on REQUIRED flag).
+#
+# Usage:
+#   zephyr_blobs_verify(MODULE module [FILES file [files...]] [REQUIRED])
+#
+# Example:
+# zephyr_blobs_verify(MODULE my_module REQUIRED) # verify all blobs in my_module and fail on error
+# zephyr_blobs_verify(MODULE my_module FILES img/file.bin) # verify a single file and print on error
+function(zephyr_blobs_verify)
+  cmake_parse_arguments(BLOBS_VERIFY "REQUIRED" "MODULE" "FILES" ${ARGN})
+
+  if(NOT DEFINED BLOBS_VERIFY_MODULE)
+    message(FATAL_ERROR "MODULE required when calling ${CMAKE_CURRENT_FUNCTION}")
+  endif()
+
+  if(NOT WEST)
+    return()
+  endif()
+
+  execute_process(
+    COMMAND ${WEST} blobs list "${BLOBS_VERIFY_MODULE}" --format "{status} {abspath}"
+    OUTPUT_VARIABLE BLOBS_LIST_OUTPUT
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    COMMAND_ERROR_IS_FATAL ANY
+  )
+
+  if(${BLOBS_VERIFY_REQUIRED})
+    set(msg_lvl FATAL_ERROR)
+  else()
+    set(msg_lvl WARNING)
+  endif()
+
+  string(REPLACE "\n" ";" BLOBS_LIST ${BLOBS_LIST_OUTPUT})
+
+  if(DEFINED BLOBS_VERIFY_FILES)
+    foreach(file ${BLOBS_VERIFY_FILES})
+      # Resolve path.
+      if(IS_ABSOLUTE ${file})
+        file(TO_NATIVE_PATH "${file}" path)
```

Comment:
Good catch that `blobs list --format "{abspath}"` is actually using platform path separators 👍 .

We can extend `blobs list` format specifier to accept `--format="{posixpath}"` and avoid this call, and another benefit of that is that it will align the function with `west list --format="{path} {abspath} {posixpath}"`.

Note, keeping a `TO_NATIVE_PATH` is also acceptable.